src/rotation.zig

```diff
@@ -821,13 +821,16 @@ pub const RotationModes = struct {
 						const rY = @intFromBool(x*sin + y*cos > 0);
 
 						if(hasSubBlock(@intCast(old), @intCast(i), @intCast(j), @intCast(k))) {
-							new |= subBlockMask(rX, rY, @intCast(k));
+							new &= ~subBlockMask(rX, rY, @intCast(k));
 						}
 					};
 					rotationTable[a][old] = new;
 				}
 			};
-			if(data >= 256) return 0;
+			if(data > 0b11111111) {
+				std.log.err("Invalid stair data: {}", .{data});
```

Comment:
I'm not sure if logging is a good idea here. Could you explain why you think it's helpful, compared to the fact that this may severely spam the console if an addon changes the rotation mode of a block?

Reply:
It was helpful when figuring out the problem in that particular case, but it is not like we desperately need it right now.
```suggestion
```

Reply:
Actually that change here could be discarded unless you agree that 0b11111111 better expresses the camp we want there.

Reply:
The 0b11111111 does make it slightly less readable in my opinion, it's just too many 1s in a row.